pkgs/tools/graphics/perceptualdiff/default.nix

```
@@ -19,6 +19,6 @@ stdenv.mkDerivation rec {
     homepage = "https://github.com/myint/perceptualdiff";
     license = licenses.gpl2Plus;
     maintainers = with maintainers; [ uri-canva ];
-    platforms = platforms.x86;
+    platforms = platforms.x86 ++ ["aarch64-linux"];
```

```suggestion
    platforms = platforms.unix;
```

---

I don't think this is the same, wouldn't this include `aarch64-darwin`? The package doesn't work there.

---

It is never supposed to work? Or it doesn't work right now?

Because if the second I think it is better to set `meta.broken = stdenv.hostPlatform.system == "aarch64-darwin"` instead.

---

The way it is set right now we are also limiting for things like `arm7v-linux`, for example.

---

The release is from 5 years ago, before `aarch64-darwin` was even announced, so I don't think it was ever supposed to work as far as upstream is concerned. I personally didn't expect it to work when I first added the derivation.

---

Let's express it like this instead then:
```suggestion
    platforms = remove "aarch64-darwin" platforms.unix;
```

---

Still, upstream still has updates. I don't think it is impossible to it eventually support `aarch64-darwin`.

Also, `aarch64-darwin` is not a Tier 1/Tier 2 supported system, so it doesn't matter too much if this doesn't build (and I think it is nice to allow people to build themselves, so they can know what is broken and maybe submit a patch later fixing the system).

---

> Let's express it like this instead then:

I would still prefer it to be marked as broken. We generally avoid manipulation of the default platforms for code that can be build from source.